contracts/mainnet/connectors/compound-import/helpers.sol

```diff
@@ -0,0 +1,218 @@
+     * @dev uses the cEth contract for ETH repays, otherwise the general cToken interface
+     * @param _userAccount the user address for which debt is to be repayed
+     * @param _cTokenContracts array containing all interfaces to the cToken contracts in which the user has debt positions
+     * @param _borrowAmts array containing the amount borrowed for each token
+     */
+    function _repayUserDebt(
+        address _userAccount,
+        CTokenInterface[] memory _cTokenContracts,
+        uint[] memory _borrowAmts
+    ) internal {
+        for(uint i = 0; i < _cTokenContracts.length; i++){
+            if(_borrowAmts[i] > 0){
+                if(address(_cTokenContracts[i]) == address(cEth)){
+                    cEth.repayBorrowBehalf{value: _borrowAmts[i]}(_userAccount);
+                }
+                else{
+                    require(_cTokenContracts[i].repayBorrowBehalf(_userAccount, _borrowAmts[i]) == 0, "repayOnBehalf-failed");
+                }
+            }
+        }
+    }
+
+    /**
+     * @notice used to transfer user's supply position on Compound to DSA
+     * @dev uses the transferFrom token in cToken contracts to transfer positions, requires approval from user first
+     * @param _userAccount address of the user account whose position is to be transferred
+     * @param _cTokenContracts array containing all interfaces to the cToken contracts in which the user has supply positions
+     * @param _amts array containing the amount supplied for each token
+     */
+    function _transferTokensToDsa(
+        address _userAccount, 
+        CTokenInterface[] memory _cTokenContracts,
+        uint[] memory _amts
+    ) internal {
+        for(uint i = 0; i < _cTokenContracts.length; i++) {
+            if(_amts[i] > 0) {
+                require(_cTokenContracts[i].transferFrom(_userAccount, address(this), _amts[i]), "ctoken-transfer-failed-allowance?");
+            }
+        }
+    }
+
+    /**
+     * @notice borrows the user's debt positions from Compound via DSA, so that its debt positions get imported to DSA
+     * @dev actually borrow some extra amount than the original position to cover the flash loan fee
+     * @param _cTokenContracts array containing all interfaces to the cToken contracts in which the user has debt positions
+     * @param _amts array containing the amounts the user had borrowed originally from Compound plus the flash loan fee
+     * @param _flashLoanFee flash loan fee (in percentage and scaled up to 10**2)
+     */
+    function _borrowDebtPosition(
+        CTokenInterface[] memory _cTokenContracts, 
+        uint256[] memory _amts,
+        uint256 _flashLoanFee
+    ) internal {
+        for (uint i = 0; i < _cTokenContracts.length; i++) {
+            if (_amts[i] > 0) {
+                require(_cTokenContracts[i].borrow(add(_amts[i], mul(_amts[i], mul(_flashLoanFee, 10**14)))) == 0, "borrow-failed-collateral?");
```

Comment:
@chinmay0402 @pradyuman-verma We can simplify down this: get flashloanFeeAmounts from frontend and add to borrow amount